src/union.ts

```diff
@@ -254,9 +253,24 @@ export class Union {
                 }
             }
         }
-        return Array.from(result).sort();
+        return this.sortedArrayFromReaddirResult(result);
+    };
+
+    private pathFromReaddirEntry = (readdirEntry: readdirEntry): string => {
+        if (readdirEntry instanceof Buffer || typeof readdirEntry === 'string') {
+            return String(readdirEntry);
+        }
+        return readdirEntry.name;
     };
 
+    private sortedArrayFromReaddirResult = (readdirResult: Map<string, readdirEntry>): readdirEntry[] => {
+        const array: readdirEntry[] = [];
+        for (const key of Array.from(readdirResult.keys()).sort()) {
+            const value = readdirResult.get(key);
+            if (value !== undefined) array.push(value);
```

Comment:
This check isn't needed because TypeScript won't let you push `undefined` into the `Map`